hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -837,14 +852,46 @@ public AbfsRestOperation getPathStatus(final String path, final boolean includeP
     abfsUriQueryBuilder.addQuery(HttpQueryParams.QUERY_PARAM_UPN, String.valueOf(abfsConfiguration.isUpnUsed()));
     appendSASTokenToQuery(path, operation, abfsUriQueryBuilder);
 
+    AbfsRestOperation op = null;
+    final URL url = createRequestUrl(path, abfsUriQueryBuilder.toString());
+    if (abfsConfiguration.getMode() == PrefixMode.BLOB){
+      op = getPathStatusBlob(path, includeProperties);
+    } else if (abfsConfiguration.getMode() == PrefixMode.DFS) {
+      op = new AbfsRestOperation(
+              AbfsRestOperationType.GetPathStatus,
+              this,
+              HTTP_METHOD_HEAD,
+              url,
+              requestHeaders);
+    }
+    if (op != null) {
+      op.execute(tracingContext);
+    }
+    return op;
+  }
+
+  public AbfsRestOperation getPathStatusBlob(final String path, final boolean includeProperties) throws AzureBlobFileSystemException {
```

Comment:
please reuse the header object instead of creating new.

Reply:
Done.